File: 335/2B.php
<?php
fscanf(STDIN, "%d %d %d %d", $x, $y, $x0, $y0);
$temp = str_split(trim(fgets(STDIN)));
$len = count($temp);
$ans = array_fill(0, $len + 1, 0);
$visit = array_fill(1, $x * $y, 0);
$step = 0;
$ans[0] = 1;
$visit[($x0-1)*$y+$y0] = 1;
foreach ($temp as $ch) {
    $step ++;
    switch ($ch) {
        case 'U':
            if ($x0 > 1) $x0 --;
            break;
        case 'R':
            if ($y0 < $y) $y0 ++;
            break;
        case 'D':
            if ($x0 < $x) $x0 ++;
            break;
        case 'L':
            if ($y0 > 1) $y0 --;
    }
    $tmp = ($x0 - 1) * $y + $y0;
    if ($visit[$tmp]==0) {
        $visit[$tmp] = 1;
        $ans[$step] = 1;
    }
}
$ans[$len] += $x * $y - array_sum($visit);
for ($i = 0;$i <= $len;$i ++) {
    if($i > 0) echo " ";
    echo $ans[$i];
}
echo "\n";
?>

File: 325/2B.c
#include <stdio.h>

int N;
int a[55];
int b[55];
int c[55];
int s1[55];
int s2[55];
int tmp, ans;
int i,j;

int main() {
    scanf("%d", &N);
    s1[0] = 0;
    s2[N-1] = 0;
    for(i=1;i<N;i++) {
        scanf("%d",a+i);
        s1[i] = s1[i-1]+a[i];
    }
    for(i=1;i<N;i++) {
        scanf("%d",b+i);
    }
    for(i=N-2;i>=0;i--) {
        s2[i] = s2[i+1] + b[i+1];
    }
    for(i=0;i<N;i++) {
        scanf("%d",c+i);
    }
    ans = 2e9;
    for(i=0;i<N;i++) {
        for(j=i+1;j<N;j++) {
            tmp = s1[i] + c[i] + s2[i] + s1[j] + s2[j] + c[j];
            if(tmp<ans) ans=tmp;
        }
    }
    printf("%d\n", ans);

    return 0;
}
